d2/src/main/java/com/linkedin/d2/balancer/strategies/relative/RelativeLoadBalancerStrategyFactory.java

```diff
@@ -0,0 +1,182 @@
+/*
+   Copyright (c) 2020 LinkedIn Corp.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package com.linkedin.d2.balancer.strategies.relative;
+
+import com.linkedin.d2.D2QuarantineProperties;
+import com.linkedin.d2.D2RelativeStrategyProperties;
+import com.linkedin.d2.D2RingProperties;
+import com.linkedin.d2.HashConfig;
+import com.linkedin.d2.HashMethod;
+import com.linkedin.d2.HttpMethod;
+import com.linkedin.d2.HttpStatusCodeRange;
+import com.linkedin.d2.HttpStatusCodeRangeArray;
+import com.linkedin.d2.balancer.config.RelativeStrategyPropertiesConverter;
+import com.linkedin.d2.balancer.event.EventEmitter;
+import com.linkedin.d2.balancer.event.NoopEventEmitter;
+import com.linkedin.d2.balancer.properties.ServiceProperties;
+import com.linkedin.d2.balancer.strategies.LoadBalancerStrategyFactory;
+import com.linkedin.d2.balancer.strategies.PartitionLoadBalancerStateListener;
+import com.linkedin.d2.balancer.util.hashing.HashFunction;
+import com.linkedin.d2.balancer.util.hashing.RandomHash;
+import com.linkedin.d2.balancer.util.hashing.URIRegexHash;
+import com.linkedin.d2.balancer.util.healthcheck.HealthCheckOperations;
+import com.linkedin.r2.message.Request;
+import com.linkedin.util.clock.Clock;
+import com.linkedin.util.clock.SystemClock;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ScheduledExecutorService;
+
+
+/**
+ * Load balancer factory for {@link RelativeLoadBalancerStrategy}.
+ */
+public class RelativeLoadBalancerStrategyFactory implements LoadBalancerStrategyFactory<RelativeLoadBalancerStrategy>
+{
+  // Default load balancer property values
+  public static final long DEFAULT_UPDATE_INTERVAL_MS = 5000L;
+  private static final double DEFAULT_UP_STEP = 0.05;
+  private static final double DEFAULT_DOWN_STEP = 0.2;
+  private static final double DEFAULT_RELATIVE_LATENCY_HIGH_THRESHOLD_FACTOR = 1.2;
+  private static final double DEFAULT_RELATIVE_LATENCY_LOW_THRESHOLD_FACTOR = 1.1;
```

Comment:
maybe too tight. 1.4/1.3?

Reply:
We will decide the value after we perform more tests/simulations